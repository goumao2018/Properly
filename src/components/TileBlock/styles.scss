.tileBlock {
  background: tomato;
  position: relative;
}

.tileBlock:before {
  content: attr(data-notification-count);
  position: absolute;
  top: 0;
  right: 0;
  line-height: 1.5em;
  display: block;
  radius: 50%;
  background: green;
}