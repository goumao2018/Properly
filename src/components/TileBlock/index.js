import React from "react";
import PropTypes from "prop-types";
import styles from "./styles.scss";

const TileBlock = ({ title, imageUrl, notificationCount}) => {
  return (
    <div className="tileBlock" data-notification-count={notificationCount === 0 ? "" : notificationCount}>
      <h1>{title}</h1>
      <img src={imageUrl} alt="Tile Image" />
    </div>
  )
}

TileBlock.propTypes = {
  title: PropTypes.string,
  imageUrl: PropTypes.string,
  notificationCount:PropTypes.number,
}

TileBlock.defaultProps = {
  title: "",
  imageUrl: "",
  notificationCount: 0,
}

export default TileBlock;

// Todo: 
// get scss running 
// try to import scss
