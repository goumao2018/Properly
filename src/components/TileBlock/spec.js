import React from 'react';
import { mount } from 'enzyme';
import TileBlock from './';

describe("<TileBlock />", () => {
  it("sets image url", () => {
    const component = mount(
      <TileBlock
        imageUrl="http://via.placeholder.com/300"
      />
    );
    expect(component.props().imageUrl).toEqual("http://via.placeholder.com/300");
    expect(component).toMatchSnapshot();
  })

  it("sets title", () => {
    const component = mount(
      <TileBlock
        title="TestBlock"
      />
    );
    expect(component.props().title).toEqual("TestBlock");
    expect(component).toMatchSnapshot();
  })
})