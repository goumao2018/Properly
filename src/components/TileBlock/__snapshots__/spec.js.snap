// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TileBlock /> sets image url 1`] = `
<TileBlock
  imageUrl="http://via.placeholder.com/300"
  title=""
>
  <div>
    <h1 />
    <img
      alt="Tile Image"
      src="http://via.placeholder.com/300"
    />
  </div>
</TileBlock>
`;

exports[`<TileBlock /> sets title 1`] = `
<TileBlock
  imageUrl=""
  title="TestBlock"
>
  <div>
    <h1>
      TestBlock
    </h1>
    <img
      alt="Tile Image"
      src=""
    />
  </div>
</TileBlock>
`;
